common/options/options.go

```diff
@@ -382,8 +383,9 @@ func (auth *Auth) IsSet() bool {
 // ShouldAskForPassword returns true if the user specifies a username flag
 // but no password, and the authentication mechanism requires a password.
 func (auth *Auth) ShouldAskForPassword() bool {
+	neverUsesPassword := []string{"MONGODB-X509", "GSSAPI", "MONGODB-OIDC"}
```

Comment:
my assumption here is go is smart enough to cache this in the static data segment. Making assumptions on go code gen quality is often disappointing, though.

Reply:
Oh I was thinking just for general legibility. I doubt the difference in speed would palpably affect a network-bound application like this … ?

Reply:
Oh yea, it doesn't matter at all, this is in the Ux! I just can't avoid taking a dig at go code gen :joy: